### PR TITLE
Fix bug and add comments

### PR DESCRIPTION
**Pull request by jok0436**
* Fix bugs in Turtle drawer (TDrawer)
* Add comments
* Add readme.doc with hyperlinks to each file

### CODE REVIEW

File: PyGameDrawer.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def draw_line(self, direction, distance):
         # except depends on the situation, may draw the line or not
         newCoords = self.getDestination(self.x, self.y, direction, distance)
         if (self.penDown):
-            pygame.draw.line(self.screen, self.penColour, [self.x, self.y], newCoords, 1)
+            pygame.draw.line(self.screen, self.penColour, [int(self.x), int(self.y)], newCoords, 1)
```
Comment (forestraindrip) [inline PyGameDrawer.py]: I integrate Tom's fixes of angle into the code

File: TDrawer.py [hunk 2]
```diff
@@ -18,33 +20,40 @@ def pen_up(self):
 
     # Because for along and down, we dont wanna draw line while we are moving
     def go_along(self, along):
+        x = turtle.xcor()
```
Comment (forestraindrip) [inline TDrawer.py]: Fix turtle go to absolute position instead of relative position

File: TDrawer.py [hunk 3]
```diff
@@ -18,33 +20,40 @@ def pen_up(self):
...
             turtle.penup()
-            turtle.setx(int(along))
+            turtle.setx(x + int(along))
             turtle.pendown()
         else:
-            turtle.setx(int(along))
+            turtle.setx(x + int(along))
 
     def go_down(self, down):
+        y = turtle.ycor()
```
Comment (forestraindrip) [inline TDrawer.py]: Fix turtle go to absolute position instead of relative position

File: TDrawer.py [hunk 4]
```diff
@@ -18,33 +20,40 @@ def pen_up(self):
...
-            turtle.sety(int(down))
+            turtle.sety(y + int(down))
             turtle.pendown()
         else:
-            turtle.sety(int(down))
+            turtle.goty(y + int(down))
 
     def draw_line(self, direction, distance):
-    # IF YOU WANT TO HAVE DYNAMIC DIRECTIONS YOU NEED TO UPDATE THIS CODE
-        direction = int(direction)
```
Comment (forestraindrip) [inline TDrawer.py]: This approach does not work well with triangle and rectangle drawing functions. So I create the "_convert_direction" function to address for good, hopefully.

**Comment by jok0436** [comment]
Lets merge Marcus's changes with the Master Branch and present that as our final work. Thanks for finding those errors Marcus!

File: TKDrawer.py [hunk 5]
```diff
@@ -33,6 +33,8 @@ def go_down(self, down):
     def draw_line(self, direction, distance):
         direction = int(direction)
         distance = int(distance)
+        if direction == 0 or direction == 180:
```
Comment (forestraindrip) [inline TKDrawer.py]: Grab from Tom's latest fix

File: TKDrawer.py [hunk 6]
```diff
@@ -66,7 +68,7 @@ def draw_rectangle(self, size):  # J
             ourDirection += 90
 
     def draw_triangle(self, size):  # M
-        ourDirection = 0
+        ourDirection = 120
```
Comment (forestraindrip) [inline TKDrawer.py]: Grab from Tom's latest fix

File: TDrawer.py [hunk 7]
```diff
@@ -53,14 +62,26 @@ def draw_circle(self, size):
...
-            ourDirection = 90
             self.draw_line(ourDirection, size)
+            ourDirection -= 90
 
     def draw_triangle(self, size):
         ourDirection = 0
         for i in range(3):
-            ourDirection = 120
             self.draw_line(ourDirection, size)
+            ourDirection -= 120
```
Comment (forestraindrip) [inline TDrawer.py]: Turtle uses counter clockwise rotation system, so the incremental angle should be reversed to be consistent with other drawers.

File: TDrawer.py [hunk 8]
```diff
@@ -53,14 +62,26 @@ def draw_circle(self, size):
     def draw_rectangle(self, size):
         ourDirection = 0
         for i in range(4):
-            ourDirection = 90
             self.draw_line(ourDirection, size)
+            ourDirection -= 90
```
Comment (forestraindrip) [inline TDrawer.py]: Turtle uses counter clockwise rotation system, so the incremental angle should be reversed to be consistent with other drawers.

**Pull request merged**